pkg/bootstrapper/bootstrapper_test.go

```diff
@@ -220,8 +219,10 @@ func TestCloudConfExtraction(t *testing.T) {
 	// Check that the --cloud-conf option value is present in the kubelet args and matches tempdir + /cloud.conf
 	cloudConfigOptValue, present := wnb.kubeletArgs["cloud-config"]
 	assert.True(t, present, "cloud-config option is not present in kubelet args")
-	assert.Equal(t, path.Join(dir, "cloud.conf"), cloudConfigOptValue,
+	assert.Equal(t, filepath.Join(dir, "cloud.conf"), cloudConfigOptValue,
 		"unexpected --cloud-config value %s", cloudConfigOptValue)
+	assert.Equalf(t, string(os.PathSeparator), string(cloudConfigOptValue[len(dir):][0]),
```

Comment:
Please look at the whole argument list has the correct separator and not just one option. Please add a similar test to the e2e too. There you would have to look at the service's binary path and ensure the correct separator is used.